library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
Also I changed semantics of "sorted" and "end" variables - and this caused all those questions above and confusion. Previously "end" pointed at the end of mapped index file and logically this code potentially does full scan of index file. Now "end" pointed to the end of a buffer, but logic stays the same - with scanning until reaching end of file or finding valid index.